lib/fulfil/response_parser.rb

```diff
@@ -55,6 +55,9 @@ def self.group(key_value_tuples)
     end
 
     def self.parse(item:)
```

Comment:
Please remind me; how are we parsing arrays that contain hash values? Are we calling `.parse` one by one? Otherwise, this will break existing behavior and we might be better off re-writing this a bit further.

Reply:
```
   def parse(result: nil, results: [])
      if result
        parse_single(result: result)
      else
        parse_multiple(results: results)
      end
    end

    def parse_single(result:)
      Fulfil::ResponseParser.parse(item: result)
    end

    def parse_multiple(results:)
      results.map { |result| Fulfil::ResponseParser.parse(item: result) }
    end

    def domain
      "https://#{@subdomain}.fulfil.io"
    end

    def base_url
      [domain, 'api', 'v2'].join('/')
    end

    def model_url(model:, id: nil, endpoint: nil)
      [base_url, 'model', model, id, endpoint].compact.join('/')
    end

    def request(endpoint:, verb: :get, **args)
      raise InvalidClientError if invalid?

      response = client.request(verb, endpoint, args)
      Fulfil::ResponseHandler.new(response).verify!

      response.parse
    rescue HTTP::Error => e
      puts e
      raise UnknownHTTPError, 'Unhandled HTTP error encountered'
    rescue HTTP::ConnectionError => e
      puts "Couldn't connect"
      raise ConnectionError, "Can't connect to #{base_url}"
    rescue HTTP::ResponseError => e
      raise ResponseError, "Can't process response: #{e}"
      []
    # If configured, the client will wait whenever the `RateLimitExceeded` exception
    # is raised. Check `Fulfil::Configuration` for more details.
    rescue RateLimitExceeded => e
      raise e unless config.retry_on_rate_limit?

      sleep config.retry_on_rate_limit_wait
      retry
    end

    def client
      client = HTTP.use(logging: @debug ? { logger: Logger.new($stdout) } : {})
      client = client.auth("Bearer #{@token}") if @token
      client.headers(@headers)
    end

    def config
      Fulfil.config
    end
  end
end
```